examples/TestRoutingGuard/can-load.spec.ts

```diff
@@ -0,0 +1,172 @@
+import { Location } from '@angular/common';
+import {
+  Component,
+  inject,
+  Injectable,
+  NgModule,
+  VERSION,
+} from '@angular/core';
+import {
+  CanLoadFn,
+  Router,
+  RouterModule,
+  RouterOutlet,
+} from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+import { from } from 'rxjs';
+import { mapTo } from 'rxjs/operators';
+
+import {
+  MockBuilder,
+  MockRender,
+  NG_MOCKS_GUARDS,
+  NG_MOCKS_ROOT_PROVIDERS,
+  ngMocks,
+} from 'ng-mocks';
+
+// A simple service simulating login check.
+// It will be replaced with its mock copy.
+@Injectable()
+class LoginService {
+  public isLoggedIn = false;
+}
+
+// The canLoad guard we want to test.
+const canLoadGuard: CanLoadFn = (route, segments) => {
+  if (route && segments && inject(LoginService).isLoggedIn) {
+    return true;
+  }
+
+  return from(inject(Router).navigate(['/login'])).pipe(mapTo(false));
+};
+
+// Another canLoad guard like in a real world example,
+// which should be removed from testing to avoid side effects on the route.
+const sideEffectCanLoadGuard: CanLoadFn = () => false;
+
+// A simple component pretending a login form.
+// It will be replaced with a mock copy.
+@Component({
+  selector: 'login',
+  template: 'login',
+})
+class LoginComponent {}
+
+// A simple component pretending a protected dashboard.
+// It will be replaced with a mock copy.
+@Component({
+  selector: 'dashboard',
+  template: 'dashboard',
+})
+class DashboardComponent {}
+
+@NgModule({
+  declarations: [DashboardComponent],
+  imports: [
+    RouterModule.forChild([
+      {
+        path: 'dashboard',
+        component: DashboardComponent,
+      },
+      {
+        path: '**',
+        redirectTo: 'dashboard',
+      },
+    ]),
+  ],
+  exports: [],
+})
+class DashboardModule {}
+
+// Definition of the routing module.
+@NgModule({
+  declarations: [LoginComponent, DashboardComponent],
+  exports: [RouterModule],
+  imports: [
+    RouterModule.forRoot([
+      {
+        component: LoginComponent,
+        path: 'login',
+      },
+      {
+        canLoad: [canLoadGuard, sideEffectCanLoadGuard],
+        path: '',
+        loadChildren: () => DashboardModule,
+      },
+    ]),
+  ],
+  providers: [LoginService],
+})
+class TargetModule {}
+
+describe('TestRoutingGuard:canLoad', () => {
+  // Because we want to test a canLoad guard, it means that we want to
```

Comment:
Update the text to resolve some minor sentence stuff.
  // Because we want to test a canLoad guard, it means that we want to
  // test it's integration with RouterModule.
  // Therefore, RouterModule and guard should be kept,
  // and the rest of the module which defines the route can be mocked.
  // To configure the RouterModule for the test,
  // RouterModule, RouterTestingModule.withRoutes([]), NG_MOCKS_ROOT_PROVIDERS
  // should be specified as the first parameter of MockBuilder (with empty routes).
  // The module with routes and the guard should be specified
  // as the second parameter of MockBuilder.
  // Then `NG_MOCKS_GUARDS` should be excluded to remove all guards,
  // and `canLoadGuard` should be kept to let you test it.